### DISCUSSION TITLE
CTEs and @*ToMany

### DISCUSSION THREAD

**Discussion by Pilpin** [Blazebit/blaze-persistence#1755] [Q&A] [answered]
Hello,

Is the annotation `@*ToMany` supposed to work with CTEs ?

I get this error `java.lang.IllegalArgumentException: The cte attribute [connected] does not exist!` when trying to query my view. I don't understand... Am i doing something wrong ?

I have an entities like so
```
@Entity
public class SomethingEntity {
    @Id
    @GeneratedValue
    private Long id;

    @OneToMany(mappedBy = "something", cascade = CascadeType.ALL)
    private Set<ConnectedThingEntity> connected = new HashSet<>();
}

@Entity
public class ConnectedThingEntity {
    @Id
    @GeneratedValue
    private Long id;
    
    @ManyToOne
    private SomethingEntity something;
}
```
A CTE
```
@Entity
@CTE
public class SomethingCTE {
    @Id
    private Long id;

    @OneToMany
    private Set<ConnectedThingEntity> connected;
}
```
A view
```
@EntityView(SomethingCTE.class)
public class SomethingView {
    Long getId();

    SortedSet<ConnectedThingView> getConnected();
}
```
And the criteriaBuilder
```
cbf.create(em, SomethingCTE .class)
                .with(SomethingCTE .class)
                .from(SomethingEntity .class, "s")
                .where("s.id").in(ids)
                .bind("id").select("s.id")
                .bind("connected").select("s.connected)
                .end();
```
I omitted the getters and setters for brevity.

Thank you !

**Comment by jwgmeligmeyling** [comment, marked as answer]
Well a `ToMany` relation would require either a join table or a join column on the target entity, so no, there's no way to make this work with a single CTE I am afraid. You would need to construct your query differently.

**Comment by Pilpin** [comment]
If anyone is wondering the same thing, I ended using a correlation mappings via entity array syntax which did exactly what I needed it to.
https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/index.html#entity-array-expression-correlation

```
@EntityView(SomethingCTE.class)
public class SomethingView {
    Long getId();

    @Mapping("ConnectedThingEntity[something.id IN VIEW(id)]")
    SortedSet<ConnectedThingView> getConnected();
}
```
